### DISCUSSION TITLE
Meta-problem: (not sure yet... ;))

### DISCUSSION THREAD

**Discussion by mariha** [OpenHospitalityNetwork/fedi-trustroots#74] [General]
Following the discussion [here](https://github.com/OpenHospitalityNetwork/fedi-trustroots/issues/16#issuecomment-935889693),
@lennartvanlaake wrote
> In terms of the C4 process, I would say it gives a bit too much power to a single developer to say that any implementation of fixing these problems would be fine. My preference would to create one issue to research all the things to be done for this to work, and then create separate "pull-request sized" issues for each unit of work that needs to be done to reach that goal.

I have a similar feeling that the way we do right now is too much ambiguous in terms of defining what is a valid problem to solve and what is not. For me there should be a change of ownership involved in the process: **the team owns a problem, and a developer owns a solution**. So we would need some way of collaboratively deciding which problems are worth working on and refining together what are acceptance criteria for a valid solutions to each of those problems. Once they ware in place, everyone would have full freedom to solve the issue within the boundaries defined by the team for it. 

Then, it would go as per C4: when a developer solves an issue, they create a PR to publish it (and pass ownership back to the team). At this point, any maintainer can approve or reject it based on the criteria the team has previously agreed on. We aim for no-ambiguity in defining what a valid solution is so there is no subjective judgement of a solution involved.

I tried to solve it earlier [here](https://github.com/OpenHospitalityNetwork/OpenHospitalityNetwork.github.io/wiki/Contributors-guidelines#paths-to-contribute), in _paths to contribute_ section (labels are out of date, btw).

**Comment by lennartvanlaake** [comment]
Love the idea! I think the issue now is to commit ourselves to spend lots of time on defining valid problems and acceptance criteria as a team :)

**Comment by weex** [comment]
> So we would need some way of collaboratively deciding which problems are worth working on

Maybe this is the unexpected thing. Before when I went into an issue tracker, my first thought was do I know how to solve this and if not I went away. If I did know, I had wonderful fantasies about implementing that solution and then went away. 😑

These days when I see an issue, I think immediately "is this the right problem" and "is this one that should be solved in this project" and before I even get there I see it's a feature request or is jumping straight to a solution which gives me an easy reason for commenting.

To address the challenge of how the team approaches issues, maybe some boilerplate in an issue template could say something like:

_Note from community maintainers: please consider if this issue addresses a [valid and accurate problem](https://rfc.zeromq.org/spec/42/#24-development-process) while or before proposing solutions._

**Reply by mariha** [reply]
Might it be that it’s a matter of “maturity” unrelated to how issues are phrased? ;) I think I am not opposing to it though, I guess if someone reported an issue there’s probably a valid problem they had. So maybe indeed I (we?) shouldn’t ask _if_ the issue is worth working on, but trying to collaboratively refine and accurately capture an issue is a real thing, no? 

Even _Development Process_ (2.4) in C4 that you referenced says: _(highlight is mine)_
> 3. The user or Contributor SHOULD write the issue by describing the problem they face or observe.
> 4. The user or Contributor SHOULD **seek consensus** on the accuracy of their observation, and **the value of solving the problem**.

I wander, how do you understand point 4? Like from practical perspective, what does _seeking consensus_ mean and how could we do it?

_Patch Requirements_ (2.3) also mentions:
> 2. A patch SHOULD be a minimal and accurate answer to exactly one identified and **agreed** problem.

An issue template that guides/educates a user about C4 and what we expect from them is a great idea. That actually feels like best, best place to inform about C4, followed by CONTRIBUTING.md guidelines. We also already have it here: #19

**Comment by lennartvanlaake** [comment]
One of my concerns is how we get issues on the correct level of detail. I definitely see the C4 process working if an issue would be on the level of "Remove the name Trustroots from the mobile login page", but I would foresee major problems if issue are so large that the solution chosen to it would have a very large impact on the overall design of the application, for instance the current "Remove branding from Trustroots" issue.

Is there some way in which we can encourgage/enforce issues ready to be "C4-implemented" to be nice, atomic, overseeable chunks instead of things that are problems in general but are not thought out well enough to be implemented directly?
